### PR TITLE
Rename `optuna.testing.integration` with `optuna.testing.pruner`

### LINKED ISSUE #3617: Move `optuna.testing.integration.DeterministicPruner` to `optuna.testing.pruner`
### What is an issue?

The `optuna.testing.integration.DeterministicPruner` is only used in the tests of `optuna.integration`, but it can widely used in the tests of other modules. It is natural to place it in the `optuna.testing.pruner`.

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Resolve #3617

## Description of the changes

As title.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@toshihikoyanase @keisuke-umezawa Could you review this PR?

File: tests/integration_tests/test_chainermn.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from optuna.storages import BaseStorage
 from optuna.storages import InMemoryStorage
 from optuna.storages import RDBStorage
-from optuna.testing.integration import DeterministicPruner
+from optuna.testing.pruner import DeterministicPruner
 from optuna.testing.sampler import DeterministicRelativeSampler
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_chainermn.py:25]: [Notes] At first glance, the module might be inconsistent with `optuna.pruners`, but I found that the other modules in `optuna.testing` including `sampler`, `distribution` and `storage` are also singular.

**Review by toshihikoyanase — APPROVED** [review]
The change sound reasonable to me. Thank you.

**Review by keisuke-umezawa — APPROVED** [review]
LGTM! In another PR, we may need to rename `sampler`, `storage` and `pruner` in testing to plural.

**Pull request merged**